### PR TITLE
[MRG] raw.set_montage should return self

### PR DESCRIPTION
**Pull request by Akshay0724**
Fixes issue: #4044 

**Changes made:**
I have added return self statement in `set_montage()` method so that it can be chained with other operations.

### CODE REVIEW

**Comment by agramfort** [comment]
please update a test.

**Comment by Akshay0724** [comment]
Please conform that does testing is needed to check the type of object
returned from this method?
Thanks

**Comment by agramfort** [comment]
yes

**Comment by Akshay0724** [comment]
I have added a test, does this is desired way of testing?

**Comment by teonbrooks** [comment]
LGTM

File: mne/channels/tests/test_montage.py [hunk 1]
```diff
@@ -428,6 +428,16 @@ def test_egi_dig_montage():
...
 
 
+@testing.requires_testing_data
+def test_set_montage():
+    dig_montage = read_dig_montage(egi=egi_dig_montage_fname)
+    raw_egi = read_raw_egi(egi_raw_fname)
+    raw = raw_egi.set_montage(dig_montage)
+    if not isinstance(raw, type(raw_egi)):
+        raise TypeError("set_montage should return object of type {} "
+                        .format(type(raw_egi)))
```
Comment (agramfort) [inline mne/channels/tests/test_montage.py]: not ok. Please update an existing test to keep testing time low.
also use assert_* functions and not Exceptions for testing. See above.

Comment (Akshay0724) [inline mne/channels/tests/test_montage.py]: OK, I have made these changes please have a look.

**Comment by agramfort** [comment]
thx @Akshay0724

**Comment by Akshay0724** [comment]
Thanks for merging.

**Pull request merged**